pkg/gofr/gofr_test.go

```diff
@@ -504,3 +504,107 @@ func Test_AddCronJob_Success(t *testing.T) {
 
 	assert.Truef(t, pass, "unable to add cron job to cron table")
 }
+
+func TestStaticHandler(t *testing.T) {
+	// Generate a public directory endpoint
+	directory := "./public"
+
+	if _, err := os.Stat(directory); err != nil {
+		err := os.Mkdir("./public", 0755)
+		if err != nil {
+			t.Errorf("Couldn't create a public directory, error: %s", err)
+		}
+	}
+	// Application Generation
+	app := New()
+
+	// Generate some files for testing
+	htmlContent := []byte("<html><head><title>Test Static File</title></head><body><p>Testing Static File</p></body></html>")
+
+	file1, errFile := os.Create("./public/index.html")
+
+	if errFile != nil {
+		t.Error("Couldn't create index.html file")
+	}
+
+	htmlContentSize, _ := file1.Write(htmlContent)
+
+	t.Cleanup(func() {
+		err := os.RemoveAll("./public")
+		if err != nil {
+			t.Log("Couldn't remove public folder")
+		}
+	})
+
+	// Run the Application and compare the fileType and fileSize
+	app.httpRegistered = true
+	app.httpServer.port = 8022
+
+	go app.Run()
+	time.Sleep(1 * time.Second)
+
+	host := "http://localhost:8022"
+
+	tests := []struct {
+		desc                       string
+		method                     string
+		path                       string
+		statusCode                 int
+		expectedBody               string
+		expectedBodyLength         int
+		expectedResponseHeaderType string
+	}{
+		{
+			desc:                       "check file content index.html",
+			method:                     http.MethodGet,
+			path:                       "/public/index.html",
+			statusCode:                 http.StatusOK,
+			expectedBodyLength:         htmlContentSize,
+			expectedResponseHeaderType: "text/html; charset=utf-8",
+			expectedBody:               string(htmlContent),
+		},
+		{
+			desc:       "check public endpoint",
+			method:     http.MethodGet,
+			path:       "/public",
+			statusCode: http.StatusNotFound,
+		},
+	}
+
+	for it, tc := range tests {
+		request, _ := http.NewRequestWithContext(context.Background(), tc.method, host+tc.path, http.NoBody)
+
+		request.Header.Set("Content-Type", "application/json")
+
+		client := http.Client{}
+
+		resp, err := client.Do(request)
+
+		bodyBytes, _ := io.ReadAll(resp.Body)
+
+		body := string(bodyBytes)
+
+		assert.NoError(t, err, "TEST[%d], Failed.\n%s", it, tc.desc)
+		assert.Equal(t, tc.statusCode, resp.StatusCode, "TEST[%d], Failed with Status Body.\n%s", it, tc.desc)
+
+		if tc.expectedBody != "" {
+			assert.Contains(t, body, tc.expectedBody, "TEST [%d], Failed with Expected Body. \n%s", it, tc.desc)
+		}
+
+		if tc.expectedBodyLength != 0 {
+			contentLength, _ := strconv.Atoi(resp.Header.Get("Content-Length"))
```

Comment:
It's simpler to compare the Get value to a _expectedBodyLength_ defined as a string not an int

And it's error prone, while you are ignoring errors here

Reply:
Or you can nuke the variable and use len(expexeBody) instead of having a variable linked to another